### PR TITLE
dev/migrate-layouts-directory-to-composition-api

### PR DESCRIPTION
**Pull request by PeterThrels**
layouts + composables for mixins

### CODE REVIEW

File: composables/useBase.ts [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    function doLogout() {
+        return stores.auth.logout().then(redirectToLogin)
+    }
+
+    function redirectToLogin() {
+        root!.$router.push('/login')
+    }
+
+    function showSuccessToast(msg?: string, title?: string) {
+        return root!.$weavrToast(msg !== undefined ? msg : 'All changes have been saved', {
```
Comment (oleggrech-t) [inline composables/useBase.ts:131]:
```suggestion
        return root!.$weavrToast(msg ? msg : 'All changes have been saved', {
```

Shouldn't tis be enough?

File: composables/useBase.ts [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+            title: title !== undefined ? title : 'Changes saved',
```
Comment (oleggrech-t) [inline composables/useBase.ts:132]:
```suggestion
            title: title ? title : 'Changes saved',
```

File: composables/useBase.ts [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            variant: 'success',
+        })
+    }
+
+    function showErrorToast(msg?: string, title?: string) {
+        return root!.$weavrToast(msg !== undefined ? msg : 'An error has occurred while saving', {
```
Comment (oleggrech-t) [inline composables/useBase.ts:138]:
```suggestion
        return root!.$weavrToast(msg ? msg : 'An error has occurred while saving', {
```

File: composables/useBase.ts [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+            title: title !== undefined ? title : 'Error',
```
Comment (oleggrech-t) [inline composables/useBase.ts:139]:
```suggestion
            title: title ? title : 'Error',
```

File: composables/useKyVerified.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    const showKybAlert = computed(() => {
+        if (!isCorporate.value) {
+            return false
+        }
+
+        if (corporate.value && corporateKyBStatus.value) {
+            return corporateKyBStatus.value !== KYBStatusEnum.APPROVED
+        }
+
+        return false
```
Comment (oleggrech-t) [inline composables/useKyVerified.ts:29]:
```suggestion
        if (corporate.value && corporateKyBStatus.value) {
            return corporateKyBStatus.value !== KYBStatusEnum.APPROVED
        }

        return false
```
Would the removal of that if statement produce the same result?

Comment (PeterThrels) [reply composables/useKyVerified.ts:29]: > Would the removal of that if statement produce the same result?

works as an early return, so computed won't even run for consumer @oleggrech-t which i think is ok

**Review by jasnvella2 — APPROVED** [review]

**Review by oleggrech-t — APPROVED** [review]

**Review by gaucj-t — APPROVED** [review]

File: layouts/auth.vue [hunk 6]
```diff
@@ -32,39 +32,36 @@
 </template>
 
 <script lang="ts">
```
Comment (gaucj-t) [inline layouts/auth.vue:34]: can we use <script setup> for all components  ?

File: layouts/dashboard.vue [hunk 7]
```diff
@@ -33,30 +33,40 @@
 </template>
 
 <script lang="ts">
```
Comment (gaucj-t) [inline layouts/dashboard.vue:35]: same here <script setup>

File: layouts/default.vue [hunk 8]
```diff
@@ -7,26 +7,27 @@
                 <b-spinner />
             </div>
         </div>
-        <!-- <app-footer /> -->
         <cookie-policy />
     </div>
 </template>
 
 <script lang="ts">
```
Comment (gaucj-t) [inline layouts/default.vue:14]: <script setup>

File: layouts/auth.vue (hunk 6 above)
Comment (PeterThrels) [reply layouts/auth.vue:34]: as we discussed, sugar will come later

**Pull request merged**